src/api.js

```diff
@@ -138,6 +138,54 @@ var schema = function(validator, options) {
   };
 };
 
+
+/**
+ * Validate query-string against query.
+ *
+ * options:
+ * {
+ *   param1: /.../,               // Reg-exp pattern
+ *   param2(val) { return "..." } // Function, returns message if invalid
+ * }
+ *
+ * Query-string options not specified in options will not be allowed. But it's
+ * optional if a request carries any query-string parameters at all.
+ */
+var queryValidator = function(options = {}) {
```

Comment:
Nit: if we're using var x = function () {} to ensure that a function is declared before use, we're not getting that guarantee since the var x is hoisted.  When it's hoisted, we actually get a weird situation that when the function is tried to be used, it is undefined and an appropriate error message is generated.

Let's stick to either let x = function() {} or function x() {}